tests/tabs.ts

```diff
@@ -0,0 +1,320 @@
+        expect(el1).to.be.an.instanceof(AalamTabs);
+    });
+    it('show', async () => {
+        const el = await fixture (`
+            <div id='parent'>
+                <aalam-tabs>
+                    <div slot="tab-title" id="t1">title1</div>
+                    <div slot="tab-body" id='b1'>body1</div>
+                    <div slot='tab-title' id='t2'>title2</div>
+                    <div slot='tab-body' id='b2'>body2</div>
+                    <div slot="tab-title" id="t3">title3</div>
+                    <div slot="tab-body" id='b3'>body3</div>
+                    <div slot='tab-title' id='t4'>title4</div>
+                    <div slot='tab-body' id='b4'>body4</div>
+                </aalam-tabs>
+            </div>
+        `);
+        let tabs = el.querySelector("aalam-tabs");
+        let title = document.querySelectorAll("[slot=tab-title]");
+        let body = document.querySelectorAll("[slot=tab-body]");
+
+        tabs.show(0);
+        expect(validate_el(tabs, 0)).to.equal(true);
+        expect(tabs._internal_fashion).to.equal('row');
+
+        tabs.show(1);
+        expect(validate_el(tabs, 1)).to.equal(true);
+
+        tabs.show(1);
+        expect(validate_el(tabs, 1)).to.equal(true);
+
+        tabs.show(3);
+        expect(validate_el(tabs, 3)).to.equal(true);
+
+        tabs.show(0);
+        expect(validate_el(tabs, 0)).to.equal(true);
+
+        tabs.show(2);
+        expect(validate_el(tabs, 2)).to.equal(true);
+
+        let t4 = tabs.querySelector("#t4");
+        t4.click();
+        expect(validate_el(tabs, 3)).to.equal(true);
+
+        let t2 = tabs.querySelector("#t2");
+        t2.click();
+        expect(validate_el(tabs, 1)).to.equal(true);
+    });
+
+    it('row-column-accordion', async () => {
+        const el = await fixture (`
+            <div id='parent'>
+                <aalam-tabs fashion="xs:accordion;m:row;xl:column">
+                    <div slot="tab-title" id="t1">title1</div>
+                    <div slot="tab-body" id='b1'>body1</div>
+                    <div slot='tab-title' class='tab-active' id='t2'>title2</div>
+                    <div slot='tab-body' id='b2'>body2</div>
+                    <div slot="tab-title" class='tab-active' id="t3">title3</div>
+                    <div slot="tab-body" id='b3'>body3</div>
+                    <div slot='tab-title' class='tab-active' id='t4'>title4</div>
+                    <div slot='tab-body' id='b4'>body4</div>
+                </aalam-tabs>
+            </div>
+        `);
+        let tabs = el.querySelector("aalam-tabs");
+        expect(validate_el(tabs, 1)).to.equal(true);
+        expect(validate_el(tabs, 2)).to.equal(false);
+        expect(validate_el(tabs, 3)).to.equal(false);
+
+        await setViewport({width:630, height:640})
+        tabs.show(2);
+        expect(validate_el(tabs, 2)).to.equal(true);
+        fashion_style(tabs);
+        expect(tabs._internal_fashion).to.equal('accordion');
```

Comment:
In addition to this can we validate with the elements as wel, example, we can check how manu slot=acc is present inside the tab's shadow root